dailystats.py

```diff
@@ -50,87 +51,91 @@ def read_stats_file(csv_file_name):
     return the_dictionary
 
 
-country_dictionary = None
-try:
-    country_dictionary = read_stats_file(INPUT_FILE_NAME)
-except Exception as exc:
-    print("Error reading daily stats file {}".format(INPUT_FILE_NAME))
-    print("\t\t{}".format(exc))
-    exit(-1)
-
-# TODO: The following logic needs to be improved
-############################################################################################################
-# Now process each region: Compute the average daily case count since the peak date and compare with the
-# value on the peak date
-# Write results a csv output file plus stdout
-############################################################################################################
-results_file = None
-try:
-    results_file = open(OUTPUT_FILE_NAME, "w")
-except Exception as exc:
-    print("Error opening output file {}".format(INPUT_FILE_NAME))
-    print("\t\t{}".format(exc))
-    exit(-2)
-
-output_header = "Region,Peak Case Count, Recent Average, Progress Factor\n"
-results_file.write(output_header)
-region_count = 0
-for region in country_dictionary:
-    region_count += 1
-    # TODO
-    # Some region names are prefixed with a double quote and contain commas which breaks the parsing logic
-    # This needs to be fixed
-    if region.startswith('"'):
-        continue
-    region_values = country_dictionary.get(region)
-    i = 0
-    peak_case_count = 0
-    recent_count = 0
-    recent_total = 0
-    recent_average = 0
-    for dategroup in region_values:
-        i += 1
-        # print("{} ==> {}".format(i, type(dategroup)))
-        if i == 1:
+def main():
+    """Entry point."""
+    country_dictionary = None

```

Comment:
All I've done here is move the module-level logic to a function - it now executes only if the module is run as a script (see below)